src/runtime/TemporalObject.cpp

```diff
@@ -179,7 +199,7 @@ std::string TemporalObject::tzComponent(ExecutionState& state, std::string& isoS
 
 TemporalObject::DateTime TemporalObject::parseValidIso8601String(ExecutionState& state, std::string isoString, const bool parseTimeZone = false)
 {
-    DateTime dateTime = { 0, 1, 1, 0, 0, 0, 0, 0, 0, "", "", "", "" };
+    DateTime dateTime = { 0, 1, 1, 0, 0, 0, 0, 0, 0, "", new TimeZone };
```

Comment:
I worry about memory leak.
To creating a `DateTime` instance, `TimeZone` is newly allocated.
And this `DateTime` is returned but its unclear that `TimeZone` is deallocated right away after its usage.